### ISSUE TITLE
Burned 1 asset but burned 100 and then after mint 50 assets 99 returned back

### ISSUE THREAD

**Issue by jurlit** [iron-fish/ironfish#3054]
### What happened?

I created 100 assets
Then I burned 1 and 0 left.

Then I created 50 assets and balance became 149

I think this bug could be related to these ones:

https://github.com/iron-fish/ironfish/issues/3048

https://github.com/iron-fish/ironfish/issues/3051

### Relevant log output

```shell
root@yura:/usr/src/app# ironfish wallet:balances
Account: default
 Asset Name    Asset Id                                                         Confirmed Balance 
 ───────────── ──────────────────────────────────────────────────────────────── ───────────────── 
 "zp-node-777" 6cc2d479e63c1065e652f8f404a75b83638d8f0f50a4e85910ff51771387b0c4 1.00000000        
 "zp-node-777" 7f0a81694222b8a996730f0c2e27e3df4b5b39a9f1b71568b710a49449fc0a47 1.99000000        
 $IRON         d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000004        
 zp-node-777   f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 100.00000000      
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:transactions
 Timestamp                Status      Type    Hash   Amount ($IRON)      Fee ($IRON)         Notes Spends Mints Burns Expiration 
 ──────────────────────── ─────────── ─────── ────── ─────────────────── ─────────────────── ───── ────── ───── ───── ────────── 
 1/21/2023 8:10:45 AM UTC confirmed   send    4b1d6… -0.00000001         0.00000001          2     1      1     0     6089       
 1/20/2023 10:09:51 AM UTC confirmed   send    b116… -0.00000001         0.00000001          2     2      0     1     4619       
 1/20/2023 9:55:08 AM UTC confirmed   send    f7d73… -0.00000001         0.00000001          3     2      0     0     4614       
 1/20/2023 9:46:27 AM UTC confirmed   send    2180e… -0.00000002         0.00000002          2     1      1     0     4607       
 1/20/2023 9:35:55 AM UTC confirmed   send    1aea3… -0.00000001         0.00000001          2     1      1     0     4605       
 1/19/2023 3:45:22 PM UTC confirmed   receive b89bd… 0.00000005          0.00000050          51    1      0     0     3513       
 1/19/2023 3:41:38 PM UTC confirmed   receive d4af6… 0.00000005          0.00000050          51    1      0     0     3507       
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish debug
Database in use, skipping output that requires database.
Stop the node and run the debug command again to show full output.

Iron Fish version               0.1.62 @ 4ab43f3
Iron Fish library               0.0.39 @ 4ab43f3
Operating system                Linux x64
CPU model(s)                    AMD EPYC 7502P 32-Core Processor
CPU threads                     40
RAM total                       70.67 GiB
Heap total                      4.00 GiB
Node version                    v18.12.1
ironfish in PATH                true
Garbage Collector Exposed       false
Telemetry enabled               true
Node name                       zp-node-777
Block graffiti                  zp-node-777
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:send --confirm --assetId=f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 --amount=1 --fee=0.00000001
Enter the public address of the recipient: ^C
root@yura:/usr/src/app# ironfish wallet:burn --confirm --assetId=f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 --amount=1 --fee=0.00000001
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Burned asset f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 from default
Value: 1.00000000

Transaction Hash: 082f31d0881add25ac64587b55b2badccdb69bcd3882f8c1a8ce5780261fd967

Find the transaction on https://explorer.ironfish.network/transaction/082f31d0881add25ac64587b55b2badccdb69bcd3882f8c1a8ce5780261fd967 (it can take a few minutes before the transaction appears in the Explorer)
root@yura:/usr/src/app# ironfish wallet:send --confirm --assetId=f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 --amount=1 --fee=0.00000001 --to=dfc2679369551e64e3950e06a88e68466e813c63b100283520045925adbe59ca
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s
An error occurred while sending the transaction.
 ›   Error: Request failed (400) insufficient-balance: Your balance is too low. Add funds to your account first
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish faucet
                                           
::::::::::          :::::::::::::::::         
::::::::::::       :::::::::::::::::::        
:::::::::::::     :::::::::::::::::::::       
::::::::::::::   ::::::::::::::::::::::::     
 ::::::::::::: ::::::::::        :::::::::    
   :::::::::::::::::::::          ::::::::::  
   :::::::::::::::::::::          :::::::::   
 ::::::::::::: ::::::::::        :::::::::    
::::::::::::::   ::::::::::::::::::::::::     
:::::::::::::     :::::::::::::::::::::       
::::::::::::       :::::::::::::::::::        
::::::::::           ::::::::::::::::         

Enter your email to stay updated with Iron Fish: 
Collecting your funds... Too many faucet requests
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:balances
Account: default
 Asset Name    Asset Id                                                         Confirmed Balance 
 ───────────── ──────────────────────────────────────────────────────────────── ───────────────── 
 "zp-node-777" 6cc2d479e63c1065e652f8f404a75b83638d8f0f50a4e85910ff51771387b0c4 1.00000000        
 "zp-node-777" 7f0a81694222b8a996730f0c2e27e3df4b5b39a9f1b71568b710a49449fc0a47 1.99000000        
 $IRON         d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000001        
 zp-node-777   f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 0.00000000        
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:mint --assetId=f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 --amount=50 --confirm --fee=0.00000001
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Minted asset zp-node-777 from default
Asset Identifier: f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2
Value: 50.00000000

Transaction Hash: 07440d64bce13777a4a8d6408d3c150166f1448c9d5bd3ed53252a911a17f740

Find the transaction on https://explorer.ironfish.network/transaction/07440d64bce13777a4a8d6408d3c150166f1448c9d5bd3ed53252a911a17f740 (it can take a few minutes before the transaction appears in the Explorer)
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# exit
exit
root@yura:~# 
root@yura:~# 
                                              

root@yura:~# docker exec -it ironfish bash
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:address
Account: default, public key: e3b401bfde5538940a3ac68b7f968084e3955b04934e596080628534d547ce0c
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:balances
Account: default
 Asset Name    Asset Id                                                         Confirmed Balance 
 ───────────── ──────────────────────────────────────────────────────────────── ───────────────── 
 "zp-node-777" 6cc2d479e63c1065e652f8f404a75b83638d8f0f50a4e85910ff51771387b0c4 1.00000000        
 "zp-node-777" 7f0a81694222b8a996730f0c2e27e3df4b5b39a9f1b71568b710a49449fc0a47 1.99000000        
 $IRON         d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000002        
 zp-node-777   f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 149.00000000      
root@yura:/usr/src/app# 
root@yura:/usr/src/app# 
root@yura:/usr/src/app# ironfish wallet:send --confirm --assetId=f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 --amount=1 --fee=0.00000001 --to=dfc2679369551e64e3950e06a88e68466e813c63b100283520045925adbe59ca
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Sending f76e7d122802064b353b881de49c3d72174374623bcbdea6dbbd104e794726c2 1.00000000 to dfc2679369551e64e3950e06a88e68466e813c63b100283520045925adbe59ca from default
Transaction Hash: 9074fb8492e99b25539c42a7e7e601d61bba60023c34f7dde50d182cec1b584f
Transaction fee: $IRON 0.00000001

Find the transaction on https://explorer.ironfish.network/transaction/9074fb8492e99b25539c42a7e7e601d61bba60023c34f7dde50d182cec1b584f (it can take a few minutes before the transaction appears in the Explorer)
root@yura:/usr/src/app# exit
```

### Graffiti

zp-node-777

**Comment by SPapkovich** [comment]
According to your actions, I bet you are facing this issue:
https://github.com/iron-fish/ironfish/issues/3035

**Comment by hughy** [comment]
Resolved by #3059
